snap/validate.go

```diff
@@ -26,6 +26,7 @@ import (
 
 // Regular expression describing correct identifiers.
 var validName = regexp.MustCompile("^[a-z](?:-?[a-z0-9])*$")
+var validEpoch = regexp.MustCompile("^(?:0|(?:[1-9][0-9]*[*]?))$")
```

Comment:
I guess I'm outing myself as a bit of  a loser here, but I found the regexp a bit hard to read initially. But doing a manual parser is also not pleasant (https://play.golang.org/p/4U738caveV) so :+1: for the regexp.